### DISCUSSION TITLE
Property innertext was not found.

### DISCUSSION THREAD

**Discussion by Mephisto090** [cefsharp/CefSharp.Dom#15] [Q&A] [answered]
Hello dear developer,

I get the error message: `CefSharp.Puppeteer.PuppeteerException: "Property innertext was not found."`

I use very simple 

```
    Public Async Function GetElementString(ByVal Elements As ElementHandle, ByVal Attribute As String) As Task(Of String)
        If Attribute.ToLower = "innertext" Or Attribute.ToLower = "outerText" Or Attribute.ToLower = "innerHTML" Or Attribute.ToLower = "outerHTML" Then
            Return Await Elements.GetPropertyValueAsync(Of String)(Attribute.ToLower)
        Else
            Return Await Elements.GetAttributeValueAsync(Of String)(Attribute.ToLower)
        End If
    End Function
```

I hope this information is enough for you.

I use an extra function because I have to distinguish between each attribute. Would it be possible for cefsharp to make a difference?

Greetings from Germany

**Comment by amaitland** [comment, marked as answer]
> CefSharp.Puppeteer.PuppeteerException: "Property innertext was not found."

The error is self explanatory really, there is no such property. The language is case sensitive, it's innerText not innertext. 

https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/innerText

> I use an extra function because I have to distinguish between each attribute

As I've said before you'd be better removing these functions they make your code harder to read and maintain, offer little value.

**Reply by amaitland** [reply]
Instead of helper methods like your getelementstring try writing some extension methods for ElementHandle, add GetInnerTextAsync etc. 

https://docs.microsoft.com/en-us/dotnet/visual-basic/programming-guide/language-features/procedures/how-to-write-an-extension-method

Your code will be easier to read and maintain

**Reply by Mephisto090** [reply]
Its work thanks.

What is this error telling me? :D
`CefSharp.Puppeteer.PuppeteerException: "Node is either not visible or not an HTMLElement"`

**Reply by Mephisto090** [reply]
> CefSharp.Puppeteer.PuppeteerException: "Property innertext was not found."

Isn't there a way you can output the error like this: `Property innertext was not found. The property must be called innerText!`

**Reply by amaitland** [reply]
That's a nice idea, doubt it would be practical to implement. 

Improving the error message reminding users that properties are case sensitive and to check their spelling is a simple improvement.

**Reply by Mephisto090** [reply]
> Improving the error message reminding users that properties are case sensitive and to check their spelling is a simple improvement.

Can you tell me what you mean by this error?

**Reply by amaitland** [reply]
Error message has been improved in commit https://github.com/cefsharp/Puppeteer/commit/107e4a5d6dab05fe8165d30fa8de9cab3fcde2b4 and will be in the next release.
